apps/src/aichat/views/ModelCustomizationWorkspace.tsx

```diff
@@ -1,21 +1,57 @@
 import React from 'react';
+import {useSelector} from 'react-redux';
 
-import Tabs from './tabs/Tabs';
+import {LabState} from '@cdo/apps/lab2/lab2Redux';
+import {AichatLevelProperties} from '../types';
+import Tabs, {Tab} from './tabs/Tabs';
 import PromptCustomization from './modelCustomization/PromptCustomization';
 import RetrievalCustomization from './modelCustomization/RetrievalCustomization';
 import PublishNotes from './modelCustomization/PublishNotes';
 import styles from './model-customization-workspace.module.scss';
+import {
+  DEFAULT_MODEL_CARD_INFO,
+  DEFAULT_PROMPT_CUSTOMIZATIONS,
+  DEFAULT_RETRIEVAL_CONTEXTS,
+} from './modelCustomization/constants';
+import {isHidden} from './modelCustomization/utils';
 
 const ModelCustomizationWorkspace: React.FunctionComponent = () => {
+  const {retrievalContexts, modelCardInfo, botName, temperature, systemPrompt} =
+    useSelector(
```

Comment:
minor: you can use our cool new `useAppSelector` here which has type info about all of the typescript redux stores, so you don't need to cast the `state` object

Reply:
Done, thanks!